### ISSUE TITLE
Check if CtPackage contains a package-info.java

### ISSUE THREAD

**Issue by msteinbeck** [INRIA/spoon#2955]
Is there a method in CtPackage which allows me to check whether a package contains a `package-info.java` file?

**Comment by nharrand** [comment]
Even if it is not super clean, you can do the following:

```java
//Assuming that p is a CtPackage
((CtElementImpl) p).getPosition().contains("package-info.java");
```

When there is no `package-info.java` file, `(CtElementImpl) p).getPosition()` returns `Unknown file`, when there is one, it returns a position in `package-info.java`.

**Comment by surli** [comment]
Actually you can check it from the compilation units:
```
CompilationUnit cu = element.getPosition().getCompilationUnit();
if (cu.getUnitType() == CtCompilationUnit.UNIT_TYPE.PACKAGE_DECLARATION) {
  ... 
}
```

**Comment by nharrand** [comment]
I think `cu` is `null` when the `package-info.java` does not exists. (But I agree it is better.)

**Comment by nharrand** [comment]
May be we could add a something like `boolean CtPackage#hasPackageInfo()` to the API.

**Comment by msteinbeck** [comment]
Similar to `package-info.java`, I need to know whether a CtPackage contains a `module-info.java`. How about adding an `isEmpty` method to CtPackage?

**Comment by nharrand** [comment]
Shouldn't `module-info.java` be associated with `CtModule` rather than `CtPackage`?
(Nitpicking) I don't really like the name `isEmpty` as I find it missleading, it could mean that the package does not contain any `CtType`. But regardless of the name, I agree that we should add the behavior.

**Comment by msteinbeck** [comment]
> I don't really like the name isEmpty as I find it missleading, it could mean that the package does not contain any CtType

That's exactly what I need. I want to know whether a package does not contain a type, package-info.java, and module-info.java (and other info files provided by future SDKs). I can't just check if the path of a package contains any *-info.java file because they might have been excluded from build, thus being not part of a model.

**Comment by msteinbeck** [comment]
Due to being a `CtNamedElement`, checking for any `CtModule` (`module-info.java`) should be feasible with `CtPackage#getElements(...)`.

**Comment by nharrand** [comment]
Ok I am sold on the `isEmpty` method then.
But, I am a bit fuzzy on modules. In my understanding, a module declares packages, but packages do not contain modules. It just happen that in most build system, package tree are reproduced in terms of directory tree. Am I missing something?

**Comment by nharrand** [comment]
I have implemented a PR ( #2957 ) containing `CtPackage#hasPackageInfo()` and `CtPackage#isEmpty`. But it does not solve your module-info problem. But I assume that something like `ctPackage.getDeclaringModule().getPosition()` could be done.

WDYT?

**Comment by nharrand** [comment]
I'll close this issue as I think #2957 solves this. Don't hesitate to reopen if I missed something.
